datalake/metadata.py

```diff
@@ -23,28 +25,35 @@ def __init__(self, *args, **kwargs):
         kwargs = deepcopy(kwargs)
         super(Metadata, self).__init__(*args, **kwargs)
         self._add_id()
+        self._ensure_version()
         self._validate()
         self._normalize_dates()
 
     def _add_id(self):
         self['id'] = uuid4().hex
 
+    def _ensure_version(self):
+        if 'version' not in self:
+            self['version'] = self._VERSION
+
     def _validate(self):
         self._validate_required_fields()
         self._validate_version()
 
-    _REQUIRED_METADATA_FIELDS = ['version', 'start', 'end', 'where', 'what']
+    _REQUIRED_METADATA_FIELDS = ['version', 'start', 'end', 'where', 'what',
+                                 'id', 'hash']
 
     def _validate_required_fields(self):
         for f in self._REQUIRED_METADATA_FIELDS:
-            if f not in self:
+            if f not in self or self[f] is None:
```

Comment:
Not terribly important, but how about [get()](https://docs.python.org/2/library/stdtypes.html#dict.get)? It'll return `None` if the key is not in the dictionary, without raising an exception.